### ISSUE TITLE
Masking?

### ISSUE THREAD

**Issue by eboatwright** [not-fl3/macroquad#367]
Is there a system for texture masking? I'm making a lighting system for my survival game. So, what I wanted to do, is have a black texture on the screen, then mask out parts of it to make parts of the screen visible. (For example the player, a campfire and more)

**Comment by willi-kappler** [comment]
Yes that would be very cool. [Notan](https://github.com/Nazariglez/notan) does this:

https://nazariglez.github.io/notan-web/examples/draw_mask_texture.html

**Comment by eboatwright** [comment]
Huh, never heard of Notan! But, I would *really* appreciate this feature :)
